vid/src/advz.rs

```diff
@@ -715,27 +689,41 @@ where
                         code_word_size,
                         &multi_open_domain,
                     )
+                    .map(|poly_evals| {
+                        assert_eq!(poly_evals.len(), code_word_size); // sanity
+                        poly_evals
+                    })
                     .map_err(vid)
                 })
                 .collect::<Result<Vec<_>, VidError>>()?;
+            assert_eq!(all_poly_evals.len(), polys.len()); // sanity
 
-            // distribute evals from each poly among the storage nodes
+            // Populate a Vec of polynomial evaluations for all storage nodes:
+            //
+            // The `i`th item is a Vec of `polys.len() * multiplicity`
+            // polynomial evaluations.
             //
-            // perf warning: runtime is quadratic in payload_size
-            let mut all_storage_node_evals = vec![Vec::with_capacity(polys.len()); code_word_size];
+            // Evaluations for storage node `i` are ordered as follows. Define
```

Comment:
Great doc!

Reply:
Thanks!

> Don't fully understand the motivation for moving to a different "share evaluation layout".
But they seem equivalent and the docs added are great and needed. Thanks.

The only "motivation" is I had to rewrite this part from scratch and this is the first thing I coded. I don't know whether there's a reason to prefer one layout over another. But if you have a reason then perhaps we could change it.